### DISCUSSION TITLE
retrieve tar package file

### DISCUSSION THREAD

**Discussion by realyukii** [msys2/MINGW-packages#22780] [Q&A] [answered]
I'm looking for a way to download and retrieve a file from installed package, so, I'm in the scenario which I want to downgrade openssh package for some reason, the `pacman -Qi openssh` says:
```
pacman -Qi openssh
Name            : openssh
Version         : 9.9p1-1
Description     : Free version of the SSH connectivity tools
Architecture    : x86_64
URL             : https://www.openssh.com/portable.html
Licenses        : custom:BSD
Groups          : net-utils
Provides        : None
Depends On      : heimdal  libedit  libxcrypt  libfido2  openssl
Optional Deps   : None
Required By     : None
Optional For    : None
Conflicts With  : None
Replaces        : None
Installed Size  : 6.18 MiB
Packager        : CI (msys2/msys2-autobuild/22f1e5ad/10962750180)
Build Date      : Fri Sep 20 23:20:55 2024
Install Date    : Sun Dec 8 23:43:40 2024
Install Reason  : Explicitly installed
Install Script  : No
Validated By    : Signature
```

`CI (msys2/msys2-autobuild/22f1e5ad/10962750180)`, is there a link to such CI? I want to downgrade to `8.9p1-3`

EDIT:
for context: so I have two machine, windows 11 act as ssh's client (host) and windows 7 act as ssh's server (virtualbox)

and the issue is that the client not receiving respond from the server, and I think maybe the culprit is compatibility issues.

I came to this conclusion after trial-and-error configuring port forwarding settings on virtual box, and it works without any problem on other services, but I'm not sure if it's the real problem so I have to confirm it by downgrade the package version

fortunately, I can find the desired package on the windows 7, it's cached and located at `/var/cache/pacman/pkg` (at this point, I have no idea how the package manager fetch it as I can't find it on [the web](http://repo.msys2.org/), maybe different msys2 version use different repository source)

time to confirm whether it's the real problem or not

EDIT: turns out there's similar question on [stackoverflow](https://stackoverflow.com/q/33969803/22382954)

**Comment by Biswa96** [comment, marked as answer]
openssh is a msys package which can be found in https://repo.msys2.org/msys/x86_64/

**Reply by Biswa96** [reply]
Downgrade may not work as intended. The dependencies have changed. You may have to recompile the old package.

**Reply by realyukii** [reply]
I have take a look at [msys2's repo](http://repo.msys2.org/) but can't find the version I'm looking for

**Reply by Biswa96** [reply]
Really old binary packages removed from repository.

**Reply by realyukii** [reply]
ah I see... thanks for the information
